CONTRIBUTING.md

```diff
@@ -9,70 +36,187 @@ license the code under the same terms as the project's overall 'outbound'
+```sh
+git clone git@github.com:YOUR_GITHUB_USER_NAME/synapse.git
+```
 
- * Please follow the [code style requirements](#code-style).
+# 4. Get in touch.
 
- * Please include a [changelog entry](#changelog) with each PR.
+Join our developer community on Matrix: #synapse-dev:matrix.org !
 
- * Please [sign off](#sign-off) your contribution.
 
- * Please keep an eye on the pull request for feedback from the [continuous
-   integration system](#continuous-integration-and-testing) and try to fix any
-   errors that come up.
+# 5. Pick an issue.
 
- * If you need to [update your PR](#updating-your-pull-request), just add new
-   commits to your branch rather than rebasing.
+Fix your favorite problem or perhaps find a [Good First Issue](https://github.com/matrix-org/synapse/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22)
+to work on.
 
-## Code style
+
+# 6. Turn coffee and documentation into code and documentation!
 
 Synapse's code style is documented [here](docs/code_style.md). Please follow
 it, including the conventions for the [sample configuration
 file](docs/code_style.md#configuration-file-format).
 
-Many of the conventions are enforced by scripts which are run as part of the
-[continuous integration system](#continuous-integration-and-testing). To help
-check if you have followed the code style, you can run `scripts-dev/lint.sh`
-locally. You'll need python 3.6 or later, and to install a number of tools:
+There is a growing amount of documentation located in the [docs](docs)
+directory. This documentation is intended primarily for sysadmins running their
+own Synapse instance, as well as developers interacting externally with
+Synapse. [docs/dev](docs/dev) exists primarily to house documentation for
+Synapse developers. [docs/admin_api](docs/admin_api) houses documentation
+regarding Synapse's Admin API, which is used mostly by sysadmins and external
+service developers.
+
+If you add new files added to either of these folders, please use [Github-Flavoured
+Markdown](https://guides.github.com/features/mastering-markdown/).
 
-```
-# Install the dependencies
-pip install -e ".[lint,mypy]"
+Some documentation also exists in [Synapse's Github
+Wiki](https://github.com/matrix-org/synapse/wiki), although this is primarily
+contributed to by community authors.
+
+
+# 7. Test, test, test!
+<a name="test-test-test"></a>
+
+While you're developing and before submitting a patch, you'll
+want to test your code.
 
-# Run the linter script
+## Run the linters.
+
+The linters look at your code and do two things:
+
+- ensure that your code follows the coding style adopted by the project;
+- catch a number of errors in your code.
+
+They're pretty fast, don't hesitate!
+
+```sh
+source ./env/bin/activate
 ./scripts-dev/lint.sh
 ```
 
-**Note that the script does not just test/check, but also reformats code, so you
-may wish to ensure any new code is committed first**.
+Note that the linters *will modify your files* to fix styling errors.
```

Comment:
No, the linters don't. The script does. (It's possible to run the linters outside the script).

```suggestion
Note that the this script *will modify your files* to fix styling errors.
```